internal/report/client.go

```diff
@@ -0,0 +1,81 @@
+package report
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	polReport "sigs.k8s.io/wg-policy-prototypes/policy-report/pkg/api/wgpolicyk8s.io/v1beta1"
+)
+
+// Save instantiates the passed namespaced PolicyReport if it doesn't exist, or
+// updated a new one if one is found
+func Save(report *polReport.PolicyReport) error {
```

Comment:
I think this should be moved into the store object. It would be more clear and also better, here we allocate a client and configure that for each policy report that we need to write; doing that inside of the Store would require just one single instance

Reply:
Done in https://github.com/kubewarden/audit-scanner/pull/40/commits/0329296a7ee8c77e295d16a18d64d32492974eee